jbpm-event-emitters/jbpm-event-emitters-elasticsearch/src/test/java/org/jbpm/event/emitters/elasticsearch/ElasticSearchEventEmitterTest.java

```diff
@@ -308,6 +308,90 @@ protected ExecutorService buildExecutorService() {
+
+            Map<String, Object> variables = new HashMap<>();
+            variables.put("initiator", "john");
+            variables.put("variable", 123);
+
+            Map<String, Object> outputVariables = new HashMap<>();
+            outputVariables.put("outcome", "good work");
+            outputVariables.put("score", 55);
+
+            when(taskData.getActivationTime()).thenReturn(sampleDate);
+            when(taskData.getActualOwner()).thenReturn(user);
+            when(taskData.getDeploymentId()).thenReturn("test");
+            when(taskData.getCreatedBy()).thenReturn(user);
+            when(taskData.getCreatedOn()).thenReturn(sampleDate);
+            when(taskData.getExpirationTime()).thenReturn(sampleDate);
+            when(taskData.getTaskInputVariables()).thenReturn(variables);
+            when(taskData.getTaskOutputVariables()).thenReturn(outputVariables);
+            when(taskData.getParentId()).thenReturn(-1L);
+            when(taskData.getProcessId()).thenReturn("process");
+            when(taskData.getProcessInstanceId()).thenReturn(99L);
+            when(taskData.isSkipable()).thenReturn(true);
+            when(taskData.getStatus()).thenReturn(Status.Reserved);
+            when(taskData.getWorkItemId()).thenReturn(100L);
+
+            when(taskInstance.getId()).thenReturn(44L);
+            when(taskInstance.getPeopleAssignments()).thenReturn(peopleAssignments);
+            when(taskInstance.getDescription()).thenReturn("simple task");
+            when(taskInstance.getFormName()).thenReturn("simpletask");
+            when(taskInstance.getName()).thenReturn("Simple Task");
+            when(taskInstance.getSubject()).thenReturn("empty");
+            when(taskInstance.getPriority()).thenReturn(5);
+            when(taskInstance.getTaskType()).thenReturn("");
+            when(taskInstance.getTaskData()).thenReturn(taskData);
+
+            List<InstanceView<?>> views = new ArrayList<>();
+
+            TaskInstanceView instanceView = new TaskInstanceView(taskInstance, null);
+            instanceView.copyFromSource();
+
+            views.add(instanceView);
+            // use latch to wait for async processing of the emitter
+            CountDownLatch latch = new CountDownLatch(1);
+            ElasticSearchEventEmitter emitter = new ElasticSearchEventEmitter() {
+
+                @Override
+                protected ExecutorService buildExecutorService() {
+                    return createExecutor(latch);
+                }
+
+            };
+            emitter.apply(views);
+            latch.await(5, TimeUnit.SECONDS);
+
+            // always close emitter to clean resources
+            emitter.close();
+
+            assertThat(responseCollector).hasSize(1);
+            assertThat(responseCollector.get(0)).isEqualToNormalizingNewlines(expectedResult);
+        } finally {
+            System.clearProperty("org.jbpm.event.emitters.elasticsearch.ignoreNull");
+        }
+    }
+    
```

Comment:
Could you include a test with initiator different than "unknown" and targetEntity not null?

Reply:
sure